pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -107,6 +107,8 @@ enum MetadataKey {
     NUM_SEGMENTS_PROCESSED("numSegmentsProcessed", MetadataValueType.INT),
     NUM_SEGMENTS_MATCHED("numSegmentsMatched", MetadataValueType.INT),
     NUM_CONSUMING_SEGMENTS_QUERIED("numConsumingSegmentsQueried", MetadataValueType.INT),
+    NUM_CONSUMING_SEGMENTS_PROCESSED("numConsumingSegmentsProcessed", MetadataValueType.INT),
```

Comment:
(MAJOR) Please revert this PR, and re-check it in after it passes the compatibility test. We cannot add `MetadataKey` in the middle because we use ordinal to encode the key. See the comment of this enum

Reply:
+1. This is a breaking change